Examples fail to build

The examples fail to build on current master (70be317ba99ddf88abd718bf6b94a9ef6408458d) on a fresh Ubuntu 20.04. VM. It seems like there have been API changes that the examples did not reflect. I suggest either fixing the issues in the examples or dropping the outdated ones and pointing people to the tests, which are a good resource as to examples of working with the library. Or perhaps some tests could be made into examples.

```c
examples/radix.c: In function ‘main’:
examples/radix.c:48:27: warning: passing argument 2 of ‘fmpz_mod_poly_init’ from incompatible pointer type [-Wincompatible-pointer-types]
   48 |     fmpz_mod_poly_init(A, m);
      |                           ^
      |                           |
      |                           fmpz * {aka long int *}
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:108:68: note: expected ‘const fmpz_mod_ctx_struct *’ {aka ‘const struct fmpz_mod_ctx *’} but argument is of type ‘fmpz *’ {aka ‘long int *’}
  108 | void fmpz_mod_poly_init(fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |                                               ~~~~~~~~~~~~~~~~~~~~~^~~
examples/radix.c:49:27: warning: passing argument 2 of ‘fmpz_mod_poly_init’ from incompatible pointer type [-Wincompatible-pointer-types]
   49 |     fmpz_mod_poly_init(B, m);
      |                           ^
      |                           |
      |                           fmpz * {aka long int *}
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:108:68: note: expected ‘const fmpz_mod_ctx_struct *’ {aka ‘const struct fmpz_mod_ctx *’} but argument is of type ‘fmpz *’ {aka ‘long int *’}
  108 | void fmpz_mod_poly_init(fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |                                               ~~~~~~~~~~~~~~~~~~~~~^~~
examples/radix.c:50:27: warning: passing argument 2 of ‘fmpz_mod_poly_init’ from incompatible pointer type [-Wincompatible-pointer-types]
   50 |     fmpz_mod_poly_init(r, m);
      |                           ^
      |                           |
      |                           fmpz * {aka long int *}
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:108:68: note: expected ‘const fmpz_mod_ctx_struct *’ {aka ‘const struct fmpz_mod_ctx *’} but argument is of type ‘fmpz *’ {aka ‘long int *’}
  108 | void fmpz_mod_poly_init(fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |                                               ~~~~~~~~~~~~~~~~~~~~~^~~
examples/radix.c:51:27: warning: passing argument 2 of ‘fmpz_mod_poly_init’ from incompatible pointer type [-Wincompatible-pointer-types]
   51 |     fmpz_mod_poly_init(t, m);
      |                           ^
      |                           |
      |                           fmpz * {aka long int *}
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:108:68: note: expected ‘const fmpz_mod_ctx_struct *’ {aka ‘const struct fmpz_mod_ctx *’} but argument is of type ‘fmpz *’ {aka ‘long int *’}
  108 | void fmpz_mod_poly_init(fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |                                               ~~~~~~~~~~~~~~~~~~~~~^~~
examples/radix.c:53:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   53 |     fmpz_mod_poly_set_coeff_ui(A, 3, 5);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:54:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   54 |     fmpz_mod_poly_set_coeff_ui(A, 4, 4);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:56:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   56 |     fmpz_mod_poly_set_coeff_ui(B, 0, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:57:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   57 |     fmpz_mod_poly_set_coeff_ui(B, 2, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:58:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   58 |     fmpz_mod_poly_set_coeff_ui(B, 3, 5);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:59:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   59 |     fmpz_mod_poly_set_coeff_ui(B, 4, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:60:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   60 |     fmpz_mod_poly_set_coeff_ui(B, 5, 5);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:61:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   61 |     fmpz_mod_poly_set_coeff_ui(B, 8, 8);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:62:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   62 |     fmpz_mod_poly_set_coeff_ui(B, 9, 8);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:63:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   63 |     fmpz_mod_poly_set_coeff_ui(B, 10, 5);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:64:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   64 |     fmpz_mod_poly_set_coeff_ui(B, 12, 6);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:65:5: error: too few arguments to function ‘fmpz_mod_poly_set_coeff_ui’
   65 |     fmpz_mod_poly_set_coeff_ui(B, 13, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:348:16: note: declared here
  348 | FLINT_DLL void fmpz_mod_poly_set_coeff_ui(fmpz_mod_poly_t poly, slong n,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:67:5: error: too few arguments to function ‘fmpz_mod_poly_pow’
   67 |     fmpz_mod_poly_pow(r, A, 3);
      |     ^~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:492:16: note: declared here
  492 | FLINT_DLL void fmpz_mod_poly_pow(fmpz_mod_poly_t rop, const fmpz_mod_poly_t op,
      |                ^~~~~~~~~~~~~~~~~
examples/radix.c:69:5: error: too few arguments to function ‘fmpz_mod_poly_scalar_mul_fmpz’
   69 |     fmpz_mod_poly_scalar_mul_fmpz(r, r, a);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:431:16: note: declared here
  431 | FLINT_DLL void fmpz_mod_poly_scalar_mul_fmpz(fmpz_mod_poly_t res,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:71:5: error: too few arguments to function ‘fmpz_mod_poly_pow’
   71 |     fmpz_mod_poly_pow(t, B, 2);
      |     ^~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:492:16: note: declared here
  492 | FLINT_DLL void fmpz_mod_poly_pow(fmpz_mod_poly_t rop, const fmpz_mod_poly_t op,
      |                ^~~~~~~~~~~~~~~~~
examples/radix.c:73:5: error: too few arguments to function ‘fmpz_mod_poly_scalar_mul_fmpz’
   73 |     fmpz_mod_poly_scalar_mul_fmpz(t, t, a);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:431:16: note: declared here
  431 | FLINT_DLL void fmpz_mod_poly_scalar_mul_fmpz(fmpz_mod_poly_t res,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:75:5: error: too few arguments to function ‘fmpz_mod_poly_add’
   75 |     fmpz_mod_poly_add(r, r, t);
      |     ^~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:403:16: note: declared here
  403 | FLINT_DLL void fmpz_mod_poly_add(fmpz_mod_poly_t res, const fmpz_mod_poly_t poly1,
      |                ^~~~~~~~~~~~~~~~~
examples/radix.c:81:34: warning: passing argument 2 of ‘fmpz_mod_poly_init’ from incompatible pointer type [-Wincompatible-pointer-types]
   81 |         fmpz_mod_poly_init(b[i], m);
      |                                  ^
      |                                  |
      |                                  fmpz * {aka long int *}
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:108:68: note: expected ‘const fmpz_mod_ctx_struct *’ {aka ‘const struct fmpz_mod_ctx *’} but argument is of type ‘fmpz *’ {aka ‘long int *’}
  108 | void fmpz_mod_poly_init(fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |                                               ~~~~~~~~~~~~~~~~~~~~~^~~
examples/radix.c:84:5: error: too few arguments to function ‘fmpz_mod_poly_randtest’
   84 |     fmpz_mod_poly_randtest(t, state, n + 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:175:16: note: declared here
  175 | FLINT_DLL void fmpz_mod_poly_randtest(fmpz_mod_poly_t f, flint_rand_t state,
      |                ^~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:88:54: error: too few arguments to function ‘fmpz_mod_poly_degree’
   88 |     flint_printf("  Degree of the radix:     %wd\n", fmpz_mod_poly_degree(r));
      |                                                      ^~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:211:7: note: declared here
  211 | slong fmpz_mod_poly_degree(const fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |       ^~~~~~~~~~~~~~~~~~~~
examples/radix.c:89:72: warning: implicit declaration of function ‘fmpz_mod_poly_modulus’; did you mean ‘fmpz_mod_poly_mul’? [-Wimplicit-function-declaration]
   89 |     flint_printf("  Bit size of the modulus: %wd\n", (slong) fmpz_bits(fmpz_mod_poly_modulus(r)));
      |                                                                        ^~~~~~~~~~~~~~~~~~~~~
      |                                                                        fmpz_mod_poly_mul
examples/radix.c:89:72: warning: passing argument 1 of ‘fmpz_bits’ makes pointer from integer without a cast [-Wint-conversion]
   89 |     flint_printf("  Bit size of the modulus: %wd\n", (slong) fmpz_bits(fmpz_mod_poly_modulus(r)));
      |                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                        |
      |                                                                        int
In file included from /home/ubuntu/test/flint2/fmpz_mod_poly.h:30,
                 from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz.h:442:49: note: expected ‘const fmpz *’ {aka ‘const long int *’} but argument is of type ‘int’
  442 | FLINT_DLL flint_bitcnt_t fmpz_bits(const fmpz_t f);
      |                                    ~~~~~~~~~~~~~^
examples/radix.c:90:54: error: too few arguments to function ‘fmpz_mod_poly_degree’
   90 |     flint_printf("  Degree of the input:     %wd\n", fmpz_mod_poly_degree(t));
      |                                                      ^~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:211:7: note: declared here
  211 | slong fmpz_mod_poly_degree(const fmpz_mod_poly_t poly, const fmpz_mod_ctx_t ctx)
      |       ^~~~~~~~~~~~~~~~~~~~
examples/radix.c:93:5: error: too few arguments to function ‘fmpz_mod_poly_radix_init’
   93 |     fmpz_mod_poly_radix_init(S, r, n + 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:1233:16: note: declared here
 1233 | FLINT_DLL void fmpz_mod_poly_radix_init(fmpz_mod_poly_radix_t D,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
examples/radix.c:100:5: error: too few arguments to function ‘fmpz_mod_poly_radix’
  100 |     fmpz_mod_poly_radix(b, t, S);
      |     ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:1241:16: note: declared here
 1241 | FLINT_DLL void fmpz_mod_poly_radix(fmpz_mod_poly_struct **B, const fmpz_mod_poly_t F,
      |                ^~~~~~~~~~~~~~~~~~~
examples/radix.c:108:5: error: too few arguments to function ‘fmpz_mod_poly_clear’
  108 |     fmpz_mod_poly_clear(A);
      |     ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:118:16: note: declared here
  118 | FLINT_DLL void fmpz_mod_poly_clear(fmpz_mod_poly_t poly,
      |                ^~~~~~~~~~~~~~~~~~~
examples/radix.c:109:5: error: too few arguments to function ‘fmpz_mod_poly_clear’
  109 |     fmpz_mod_poly_clear(B);
      |     ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:118:16: note: declared here
  118 | FLINT_DLL void fmpz_mod_poly_clear(fmpz_mod_poly_t poly,
      |                ^~~~~~~~~~~~~~~~~~~
examples/radix.c:110:5: error: too few arguments to function ‘fmpz_mod_poly_clear’
  110 |     fmpz_mod_poly_clear(r);
      |     ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:118:16: note: declared here
  118 | FLINT_DLL void fmpz_mod_poly_clear(fmpz_mod_poly_t poly,
      |                ^~~~~~~~~~~~~~~~~~~
examples/radix.c:111:5: error: too few arguments to function ‘fmpz_mod_poly_clear’
  111 |     fmpz_mod_poly_clear(t);
      |     ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:118:16: note: declared here
  118 | FLINT_DLL void fmpz_mod_poly_clear(fmpz_mod_poly_t poly,
      |                ^~~~~~~~~~~~~~~~~~~
examples/radix.c:116:9: error: too few arguments to function ‘fmpz_mod_poly_clear’
  116 |         fmpz_mod_poly_clear(b[i]);
      |         ^~~~~~~~~~~~~~~~~~~
In file included from examples/radix.c:24:
/home/ubuntu/test/flint2/fmpz_mod_poly.h:118:16: note: declared here
  118 | FLINT_DLL void fmpz_mod_poly_clear(fmpz_mod_poly_t poly,
      |                ^~~~~~~~~~~~~~~~~~~

```

--- a/examples/fmpz_mod_poly.c
+++ b/examples/fmpz_mod_poly.c
@@ -23,18 +23,21 @@
 int main(int argc, char* argv[])
 {
     fmpz_t n;
+    fmpz_mod_ctx_t ctx;
     fmpz_mod_poly_t x, y;
 
     fmpz_init_set_ui(n, 7);
-    fmpz_mod_poly_init(x, n);
-    fmpz_mod_poly_init(y, n);
-    fmpz_mod_poly_set_coeff_ui(x, 3, 5);
-    fmpz_mod_poly_set_coeff_ui(x, 0, 6);
-    fmpz_mod_poly_sqr(y, x);
-    fmpz_mod_poly_print(x); flint_printf("\n");
-    fmpz_mod_poly_print(y); flint_printf("\n");
-    fmpz_mod_poly_clear(x);
-    fmpz_mod_poly_clear(y);
+    fmpz_mod_ctx_init(ctx, n);
+    fmpz_mod_poly_init(x, ctx);
+    fmpz_mod_poly_init(y, ctx);
+    fmpz_mod_poly_set_coeff_ui(x, 3, 5, ctx);
+    fmpz_mod_poly_set_coeff_ui(x, 0, 6, ctx);
+    fmpz_mod_poly_sqr(y, x, ctx);
+    fmpz_mod_poly_print(x, ctx); flint_printf("\n");
+    fmpz_mod_poly_print(y, ctx); flint_printf("\n");
+    fmpz_mod_poly_clear(x, ctx);
+    fmpz_mod_poly_clear(y, ctx);
+    fmpz_mod_ctx_clear(ctx);
     fmpz_clear(n);
 
     return EXIT_SUCCESS;

--- a/examples/radix.c
+++ b/examples/radix.c
@@ -33,6 +33,7 @@ int main(void)
 
     slong i;
     fmpz_t a, m;
+    fmpz_mod_ctx_t ctx;
     fmpz_mod_poly_t A, B, r, t;
     fmpz_mod_poly_radix_t S;
     fmpz_mod_poly_struct **b;
@@ -45,59 +46,60 @@ int main(void)
     fmpz_set_ui(m, 17);
     fmpz_pow_ui(m, m, 26);
 
-    fmpz_mod_poly_init(A, m);
-    fmpz_mod_poly_init(B, m);
-    fmpz_mod_poly_init(r, m);
-    fmpz_mod_poly_init(t, m);
+    fmpz_mod_ctx_init(ctx, m);
+    fmpz_mod_poly_init(A, ctx);
+    fmpz_mod_poly_init(B, ctx);
+    fmpz_mod_poly_init(r, ctx);
+    fmpz_mod_poly_init(t, ctx);
 
-    fmpz_mod_poly_set_coeff_ui(A, 3, 5);
-    fmpz_mod_poly_set_coeff_ui(A, 4, 4);
+    fmpz_mod_poly_set_coeff_ui(A, 3, 5, ctx);
+    fmpz_mod_poly_set_coeff_ui(A, 4, 4, ctx);
 
-    fmpz_mod_poly_set_coeff_ui(B, 0, 1);
-    fmpz_mod_poly_set_coeff_ui(B, 2, 1);
-    fmpz_mod_poly_set_coeff_ui(B, 3, 5);
-    fmpz_mod_poly_set_coeff_ui(B, 4, 1);
-    fmpz_mod_poly_set_coeff_ui(B, 5, 5);
-    fmpz_mod_poly_set_coeff_ui(B, 8, 8);
-    fmpz_mod_poly_set_coeff_ui(B, 9, 8);
-    fmpz_mod_poly_set_coeff_ui(B, 10, 5);
-    fmpz_mod_poly_set_coeff_ui(B, 12, 6);
-    fmpz_mod_poly_set_coeff_ui(B, 13, 1);
+    fmpz_mod_poly_set_coeff_ui(B, 0, 1, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 2, 1, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 3, 5, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 4, 1, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 5, 5, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 8, 8, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 9, 8, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 10, 5, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 12, 6, ctx);
+    fmpz_mod_poly_set_coeff_ui(B, 13, 1, ctx);
 
-    fmpz_mod_poly_pow(r, A, 3);
+    fmpz_mod_poly_pow(r, A, 3, ctx);
     fmpz_set_ui(a, 4);
-    fmpz_mod_poly_scalar_mul_fmpz(r, r, a);
+    fmpz_mod_poly_scalar_mul_fmpz(r, r, a, ctx);
 
-    fmpz_mod_poly_pow(t, B, 2);
+    fmpz_mod_poly_pow(t, B, 2, ctx);
     fmpz_set_ui(a, 27);
-    fmpz_mod_poly_scalar_mul_fmpz(t, t, a);
+    fmpz_mod_poly_scalar_mul_fmpz(t, t, a, ctx);
 
-    fmpz_mod_poly_add(r, r, t);
+    fmpz_mod_poly_add(r, r, t, ctx);
 
     b = flint_malloc((N + 1) * sizeof(fmpz_mod_poly_struct *));
     for (i = 0; i <= N; i++)
     {
         b[i] = flint_malloc(sizeof(fmpz_mod_poly_struct));
-        fmpz_mod_poly_init(b[i], m);
+        fmpz_mod_poly_init(b[i], ctx);
     }
 
-    fmpz_mod_poly_randtest(t, state, n + 1);
+    fmpz_mod_poly_randtest(t, state, n + 1, ctx);
 
     flint_printf("Radix conversion\n");
     flint_printf("----------------\n");
-    flint_printf("  Degree of the radix:     %wd\n", fmpz_mod_poly_degree(r));
-    flint_printf("  Bit size of the modulus: %wd\n", (slong) fmpz_bits(fmpz_mod_poly_modulus(r)));
-    flint_printf("  Degree of the input:     %wd\n", fmpz_mod_poly_degree(t));
+    flint_printf("  Degree of the radix:     %wd\n", fmpz_mod_poly_degree(r, ctx));
+    flint_printf("  Bit size of the modulus: %wd\n", (slong) fmpz_bits(m));
+    flint_printf("  Degree of the input:     %wd\n", fmpz_mod_poly_degree(t, ctx));
 
     c0 = clock();
-    fmpz_mod_poly_radix_init(S, r, n + 1);
+    fmpz_mod_poly_radix_init(S, r, n + 1, ctx);
     c1 = clock();
     c  = (double) (c1 - c0) / CLOCKS_PER_SEC;
 
     flint_printf("  Precomputation:          %fs\n", c);
 
     c0 = clock();
-    fmpz_mod_poly_radix(b, t, S);
+    fmpz_mod_poly_radix(b, t, S, ctx);
     c1 = clock();
     c  = (double) (c1 - c0) / CLOCKS_PER_SEC;
 
@@ -105,21 +107,21 @@ int main(void)
 
     fmpz_clear(a);
     fmpz_clear(m);
-    fmpz_mod_poly_clear(A);
-    fmpz_mod_poly_clear(B);
-    fmpz_mod_poly_clear(r);
-    fmpz_mod_poly_clear(t);
+    fmpz_mod_poly_clear(A, ctx);
+    fmpz_mod_poly_clear(B, ctx);
+    fmpz_mod_poly_clear(r, ctx);
+    fmpz_mod_poly_clear(t, ctx);
     fmpz_mod_poly_radix_clear(S);
 
     for (i = 0; i <= N; i++)
     {
-        fmpz_mod_poly_clear(b[i]);
+        fmpz_mod_poly_clear(b[i], ctx);
         flint_free(b[i]);
     }
     flint_free(b);
+    fmpz_mod_ctx_clear(ctx);
 
     flint_randclear(state);
 
     return EXIT_SUCCESS;
 }
-

